src/AppBundle/Controller/MembershipController.php

```diff
@@ -144,9 +146,40 @@ public function pinInterestsAction(Request $request): Response
      * @Route("/inscription/choisir-des-comites", name="app_membership_choose_nearby_committee")
      * @Method("GET|POST")
      */
-    public function chooseNearByCommitteeAction(): Response
+    public function chooseNearByCommitteeAction(Request $request): Response
     {
-        return $this->render('membership/choose_nearby_committee.html.twig');
+        $membershipUtils = $this->get('app.membership_utils');
+
+        if (!$id = $membershipUtils->getNewAdherentId()) {
+            throw $this->createNotFoundException('The adherent has not been successfully redirected from the registration page.');
+        }
+
+        $manager = $this->getDoctrine()->getManager();
+
+        if (!$adherent = $manager->getRepository(Adherent::class)->find($id)) {
+            throw $this->createNotFoundException('New adherent id not found.');
+        }
+
+        $form = $this->createForm(MembershipChooseNearbyCommitteeType::class)
+            ->add('submit', SubmitType::class, ['label' => 'Terminer'])
+            ->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            foreach ($form->get('committees')->getData() as $uuid) {
+                if ($committee = $manager->getRepository(Committee::class)->findOneBy(['uuid' => $uuid])) {
+                    $manager->persist(CommitteeMembership::createForAdherent($adherent, $committee));
+                }
```

Comment:
Hence, my remark saying that we should an error in an `else`, there can be a race condition if the committee is deleted in between.